src/test/java/org/junitpioneer/jupiter/resource/ResourcesParallelismTests.java

```diff
@@ -0,0 +1,287 @@
+			failIfExecutedInParallel("test2Iteration" + iteration);
+		}
+
+		@ParameterizedTest
+		@ValueSource(ints = { 1, 2, 3 })
+		void test3(@SuppressWarnings("unused") int iteration,
+				@SuppressWarnings("unused") @Shared(factory = TemporaryDirectory.class, name = SHARED_RESOURCE_C_NAME) Path directoryC,
+				@SuppressWarnings("unused") @Shared(factory = TemporaryDirectory.class, name = SHARED_RESOURCE_A_NAME) Path directoryA)
+				throws Exception {
+			failIfExecutedInParallel("test3Iteration" + iteration);
+		}
+
+	}
+
+	static class ThrowIfTestClassConstructorsRunInParallelTestCases1 {
+
+		ThrowIfTestClassConstructorsRunInParallelTestCases1(
+				// we don't actually use the resources, we just have them injected to verify whether sharing the
+				// same resources prevent the test constructors from running in parallel
+				@SuppressWarnings("unused") @Shared(factory = TemporaryDirectory.class, name = SHARED_RESOURCE_A_NAME) Path directoryA,
+				@SuppressWarnings("unused") @Shared(factory = TemporaryDirectory.class, name = SHARED_RESOURCE_B_NAME) Path directoryB)
+				throws Exception {
+			failIfExecutedInParallel("testConstructor1");
+		}
+
+		@Test
+		void fakeTest() {
+		}
+
+	}
+
+	static class ThrowIfTestClassConstructorsRunInParallelTestCases2 {
+
+		ThrowIfTestClassConstructorsRunInParallelTestCases2(
+				@SuppressWarnings("unused") @Shared(factory = TemporaryDirectory.class, name = SHARED_RESOURCE_B_NAME) Path directoryB,
+				@SuppressWarnings("unused") @Shared(factory = TemporaryDirectory.class, name = SHARED_RESOURCE_C_NAME) Path directoryC)
+				throws Exception {
+			failIfExecutedInParallel("testConstructor2");
+		}
+
+		@Test
+		void fakeTest() {
+		}
+
+	}
+
+	static class ThrowIfTestClassConstructorsRunInParallelTestCases3 {
+
+		ThrowIfTestClassConstructorsRunInParallelTestCases3(
+				@SuppressWarnings("unused") @Shared(factory = TemporaryDirectory.class, name = SHARED_RESOURCE_C_NAME) Path directoryC,
+				@SuppressWarnings("unused") @Shared(factory = TemporaryDirectory.class, name = SHARED_RESOURCE_A_NAME) Path directoryA)
+				throws Exception {
+			failIfExecutedInParallel("testConstructor3");
+		}
+
+		@Test
+		void fakeTest() {
+		}
+
+	}
+
+	// this method is written to fail if it is executed at overlapping times in different threads
+	private static void failIfExecutedInParallel(String testName) throws InterruptedException {
+		boolean wasZero = COUNTER.compareAndSet(0, 1);
+		System.out.println(testName + ": wasZero = " + wasZero);
+		assertThat(wasZero).isTrue();
+		// wait for the next test to catch up and potentially fail
+		Thread.sleep(TIMEOUT_MILLIS);
+		boolean wasOne = COUNTER.compareAndSet(1, 2);
+		System.out.println(testName + ": wasOne = " + wasOne);
+		assertThat(wasOne).isTrue();
+		// wait for the last test to catch up and potentially fail
+		Thread.sleep(TIMEOUT_MILLIS);
+		boolean wasTwo = COUNTER.compareAndSet(2, 0);
+		System.out.println(testName + ": wasTwo = " + wasTwo);
```

Comment:
The `System.out.println` appear weird to me. Are they needed?

Reply:
They're only there for debugging purposes. I get the impression that you'd prefer they be removed? I've tentatively removed them in commit [`4c0936d`](https://github.com/junit-pioneer/junit-pioneer/pull/491/commits/4c0936dde7462855fce94d725419fc24659e8c1e), but please do let me know if you'd like me to revert it.

Reply:
Yes, I prefer them removed. Thanks!